app/controllers/EtmpCheckController.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+import models.CheckRegimeModel
+import play.api.Logger
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import services.EtmpRegimeService
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class EtmpCheckController @Inject()(cc: ControllerComponents,
+                                     etmpRegimeService: EtmpRegimeService) extends BackendController(cc) {
+
+
+  def checkEtmp(): Action[AnyContent] = Action.async { implicit request =>
+    val feJson = request.body.asJson.get
```

Comment:
As the asOpt handles the CheckRegimeModel matching, you're referring to two related, but different things here, I think. `asJson.get` - *can* be error checked, but realistically if anything calls this without sending JSON data I'd be extremely surprised.

The other line below it checks the contents of said posted JSON data for soundness, which I think is fine and a much more likely thing to fail - we already have the required safety for this and I feel that's reasonable

Reply:
Fair, it's not how I would do this but I'll demote the rank down to AMBER, so I won't block the merge based on this.

Reply:
I would think adding a recover block here so that we can fail a bit more gracefully here would be a good addition.

Reply:
It's probably as handled as above so I don't believe it's necessary - if the service is throwing unchecked exceptions (which it shouldn't be, it has a recover block within `checkETMPApi`), it can be handled beforehand.